msnoise/s03compute_cc.py

```diff
@@ -277,13 +277,24 @@ def main():
                     t1 = t1.select(component=components[0])
                     t2 = t2.select(component=components[1])
                 else:
-                    logging.debug('Checking if traces are the same length...')
-                    t1 = make_same_length(t1.copy())   
-                    t2 = make_same_length(t2.copy())                
-                    logging.debug('Traces are the same length now.')
-                    t1 = t1.rotate("NE->RT", cplAz).select(component=components[0])
-                    t2 = t2.rotate("NE->RT", cplAz).select(component=components[1])
-
+                    logging.debug('Rotating streams, making sure they are aligned')
+                    t1_novert = t1.copy()
+                    t2_novert = t2.copy()
+                    # Make temporary streams withouth the vertical components
+                    for tr in t1_novert.select(component="Z"):
+                        t1_novert.remove(tr)
+                    for tr in t2_novert.select(component="Z"):
+                        t2_novert.remove(tr)
+                    # Make these streams contain the same gaps
+                    t1_novert = make_same_length(t1_novert)   
+                    t2_novert = make_same_length(t2_novert) 
+                    # Rotate
+                    t1_novert = t1_novert.rotate("NE->RT", cplAz).select(component=components[0])
+                    t2_novert = t2_novert.rotate("NE->RT", cplAz).select(component=components[1])
+                    # Include the vertical channels again
+                    t1 = t1_novert+t1.select(component="Z")
+                    t2 = t2_novert+t2.select(component="Z")
```

Comment:
I'd rather have a first check if the components[0] or [1] actually need to be rotated = if they are Z, no need to check their length or to rotate them, if in the end the select() takes the Z component.

```python
if components[0] != Z:
    # do the clever things to rotate t1
if components[1] != Z:
    # do the clever things to rotate t2
```